src/main/resources/templates/task.html

```diff
@@ -1,59 +1,60 @@
 <!DOCTYPE html>
 <html lang="en" xmlns="http://www.w3.org/1999/xhtml" xmlns:th="http://www.thymeleaf.org">
-<head th:include="base :: header"></head>
-
-<body>
-
-<div class="container">
-    <nav th:include="navbar :: navigation"></nav>
-    <p class="lead" th:inline="text">Challenge: [[${challengename}]]</p>
-    <p th:text="${challengedesc}"></p>
-    <div class="row">
-        <div class="col-md-4">
-            <p th:inline="text">Task: [[${taskname}]]</p><br/>
-            <p th:inline="text">Level: [[${difficulty}]]</p><br/>
+    <head th:include="base :: header"></head>
+    <body> 
+        <div class="container">
+            <nav th:include="navbar :: navigation"></nav>
+            <p class="lead" th:inline="text">Challenge: [[${challengename}]]</p>
+            <p th:text="${challengedesc}"></p>
+            <div class="row">
+                <div class="col-md-4">
+                    <p th:inline="text">Task: [[${taskname}]]</p><br/>
+                    <p th:inline="text">Level: [[${difficulty}]]</p><br/>
+                </div>
+                <div class="col-md-6">
+                    <p th:text="${taskdesc}"></p><br/><br/>
+                </div>
+            </div>
+            <div th:if="${errors}" class="row">
+                <p>The following errors were encountered:</p>
+                <ul>
+                    <li th:each="error : ${errors}">
+                        <span th:text="${error}"></span>
+                    </li>
+                </ul>
+            </div>
+            <ul class="nav nav-tabs">        
+                <li th:each="editor, iter: ${editorContents}" th:class="${iter.index == 0}?'active'"><a data-toggle="tab" th:text="${editor.value.title}" th:href="'#'+${editor.key}+'_item'">Placeholder editors</a></li>        
+            </ul>
+            <div class="tab-content" th:inline="text">        
+                <div th:each="editor, iter: ${editorContents}" th:id="${editor.key}+'_item'" th:class="${iter.index == 0}?'panel panel-default tab-pane fade in active':'panel panel-default tab-pane fade'">
+                    <div class="panel-heading">
+                        <h3 class="panel-title" th:inline="text">[[${editor.value.title}]]</h3>
+                    </div>
+                    <div class="panel-body">
+                        <form method="post" action="/task">
```

Comment:
This form should be moved outside of the loop, because now each tab will have its own form and they won't necessarily be in sync when submitting.